src/Repository/RepoFileManager.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2016 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "IO/BufferedOutputStream.hxx"
+#include "IO/FileOutputStream.hxx"
+#include "IO/KeyValueFileReader.hpp"
+#include "IO/KeyValueFileWriter.hpp"
+#include "LocalPath.hpp"
+#include "IO/FileLineReader.hpp"
+#include "Net/HTTP/DownloadManager.hpp"
+#include "OS/FileUtil.hpp"
+#include "Repository/RepoFileManager.hpp"
+#include "Repository/FileRepository.hpp"
+#include "Repository/Parser.hpp"
+#include "Time/BrokenDate.hpp"
+#include "Time/BrokenDateTime.hpp"
+#include "Util/ConvertString.hpp"
+#include "Util/HexString.hpp"
+
+#define LOCAL_HASHES_FILE "local_hashes"
```

Comment:
No macros.
And I don't like the file format. It's undocumented, but doesn't contain any management information but the file name and the digest.
No last-known file size and time stamp (for quick comparison), for example.
I've been thinking about integrating SQLite in XCSoar so we can put everything in a database...

Reply:
> No macros.

constexpr instead?

> No last-known file size and time stamp (for quick comparison), for example.

I can put those into a CSV file and write some explanatory comments, if that's better.

> I've been thinking about integrating SQLite in XCSoar so we can put everything in a database...

I'd be a fan of that.